RULES_DESCRIPTIONS.md

```diff
@@ -0,0 +1,445 @@
+```
+
+## argument-limit
+
+_Description_: Warns when a function receives more parameters than the maximum set by the rule's configuration.
+Enforcing a maximum number of parameters helps keeping code readable and maintainable.
+
+_Configuration_: (int) the maximum number of parameters allowed per function.
+
+Example:
+
+```toml
+[argument-limit]
+  arguments =[4]
+```
+
+## atomic
+
+_Description_: Check for common mistaken usages of the `sync/atomic` package
+
+_Configuration_: N/A
+
+## blank-imports
+
+_Description_: Blank import should be only in a main or test package, or have a comment justifying it; this rule warns if that is not the case.
+
+_Configuration_: N/A
+
+## bool-literal-in-expr
+
+_Description_: Using Boolean literals (`true`, `false`) in logic expressions may make the code less readable. This rule
+suggests removing Boolean literals from logic expressions.
+The rule will also try to spot logical expressions that evaluate always to the same value.
+
+_Configuration_: N/A
+
+## confusing-naming
+
+_Description_: Methods or fields of `struct`s that have similar names may be misleading.
+Warns on methods and fields with names that differ only by capitalization.
+
+_Configuration_: N/A
+
+## confusing-results
+
+_Description_: Function or methods that return multiple, no named, values of the same type may induce to error. This rule warns on those cases.
+
+_Configuration_: N/A
+
+## constant-logical-expr
+
+_Description_: Using Boolean literals (`true`, `false`) in logic expressions may make the code less readable. This rule
+suggests removing Boolean literals from logic expressions.
+The rule will also try to spot logical expressions that evaluate always to the same value.
+
+_Configuration_: N/A
+
+## context-as-argument
+
+_Description_: Is a language [convention](https://github.com/golang/go/wiki/CodeReviewComments#contexts) that `context.Context` should be the first parameter of a function. This rule spots function declarations that do not follow the convention.
+
+_Configuration_: N/A
+
+## context-keys-type
+
+_Description_: Basic types should not be used as key in `context.WithValue`.
+This rule warns when that is the case.
+
+_Configuration_: N/A
+
+## cyclomatic
+
+_Description_: [Cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) is a measure of code complexity.
+Enforcing a maximum complexity per function helps keeping code readable and maintainable.
+
+_Configuration_: (int) the maximum function complexity
+
+Example:
+```toml
+[cyclomatic]
+  arguments =[3]
+```
+
+## deep-exit
+
+_Description_: Packages exposing functions that can stop program execution by exiting are hard to reuse. This rule looks for program exits in functions other than `main()` or `init()`.
+
+_Configuration_: N/A
+
+## dot-imports
+ 
+ _Description_: Importing with `.` makes the programs much harder to understand because it is unclear where names belongs to the current package or to an imported package.
```

Comment:
```suggestion
 _Description_: Importing with `.` makes the programs much harder to understand because it is unclear where names belong to the current package or to an imported package.
```